docs/setup/independent/troubleshooting-kubeadm.md

```diff
@@ -181,3 +181,37 @@ If you're using flannel as the pod network inside vagrant, then you will have to
 Vagrant typically assigns two interfaces to all VMs. The first, for which all hosts are assigned the IP address `10.0.2.15`, is for external traffic that gets NATed.
 
 This may lead to problems with flannel. By default, flannel selects the first interface on a host. This leads to all hosts thinking they have the same public IP address. To prevent this issue, pass the `--iface eth1` flag to flannel so that the second interface is chosen.
+
+### Routing errors
+
+In some situations `kubectl logs` and `kubectl run` commands may return with the following errors despite an otherwise apparently correctly working cluster:
+
+```
+Error from server: Get https://10.19.0.41:10250/containerLogs/default/mysql-ddc65b868-glc5m/mysql: dial tcp 10.19.0.41:10250: getsockopt: no route to host
+```
+
+This is due to Kubernetes using an IP that can not communicate with other IPs on the seemigly same subnet, possibly by policy of the machine provider. As an example, Digital Ocean assigns a public IP to `eth0` as well as a private one to be used internally as anchor for their floating IP feature, yet `kubelet` will pick the latter as the node's `InternalIP` instead of the public one.
+
+To check for such a scenario, using `ip addr show` is advised instead of `ifconfig` to check assigned IPs against the one in the error, as the latter will not display the offending alias IP address. Alternatively an API endpoint specific to Digital Ocean allows to query for the anchor IP from the droplet:
```

Comment:
while technical correct, this is confusing in it's logic - can it be reworded to something like:
>Use `ip addr show` to check for this scenario instead of `ifconfig` because `ifconfig` will not display the offending alias IP address.

Reply:
Much better indeed. Fixed.